python/python/lance/dataset.py

```diff
@@ -1369,11 +1374,12 @@ def create_index(
         kwargs["metric_type"] = metric
 
         index_type = index_type.upper()
-        if index_type not in ["IVF_PQ", "DISKANN"]:
+        if index_type not in ["IVF_PQ", "DISKANN", "IVF_HNSW_PQ"]:
             raise NotImplementedError(
-                f"Only IVF_PQ or DiskANN index_types supported. Got {index_type}"
+                f"Only [IVF_PQ, IVF_HNSW_PQ, DiskANN] index types supported. "
+                f"Got {index_type}"
             )
-        if index_type == "IVF_PQ":
+        if index_type.startswith("IVF"):
```

Comment:
Do we check PQ related parameters correctly?

Reply:
we do, divide the checks into two blocks